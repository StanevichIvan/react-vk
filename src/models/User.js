export default class User {
    constructor(item) {
        this.firstName = item.first_name || '';
        this.lastName = item.last_name || '';
        this.photo = item.photo_50 || '';
        this.lastSeen = item.last_seen || '';
        this.nickname = item.nickname || '';
        this.id = item.uid || 0;
    }
}