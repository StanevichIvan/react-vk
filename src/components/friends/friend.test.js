import renderer from 'react-test-renderer';
import Friend from './friend';
import * as React from "react";
import User from "../../models/User";

describe('Friend component', () => {

    it('Сorrectly rendering', () => {
        const obj = {
            first_name: "Игнат",
            last_name: "Игнатов",
            photo_50: "https://d75ljpmh88vxs.cloudfront.net/web/images/lens_flare_inspiration_square.jpg",
            last_seen: Date.now() / 1000,
            uid: 2131231
        };

        const user = new User(obj);
        const rendered = renderer.create(<Friend user={user}/>);
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});