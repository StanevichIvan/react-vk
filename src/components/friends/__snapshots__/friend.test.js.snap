exports[`Friend component Сorrectly rendering 1`] = `
<div
  className="conversation__message new"
  onClick={[Function]}>
  <img
    alt=""
    className="conversation__avatar"
    src="https://d75ljpmh88vxs.cloudfront.net/web/images/lens_flare_inspiration_square.jpg" />
  <div
    className="conversation__message-info">
    <h4
      className="conversation__name">
      Игнат
       
      Игнатов
    </h4>
  </div>
  <div
    className="conversation__message-info">
    <h4
      className="conversation__name conversation__name_right">
      <span
        className="conversation__message-count" />
      <span
        className="conversation__message-time" />
    </h4>
    <p
      className="conversation__message-text conversation__name_right">
      <i
        className="conversation__attachment" />
    </p>
  </div>
</div>
`;
